### ISSUE TITLE
Training EQTransformer: Output of predictions are tuples and not tensors

### ISSUE THREAD

**Issue by sfpoveda** [seisbench/seisbench#110]
Hi,

Trying to train EQTransformer model I found a problem. In the code line 'pred = model(batch["X"].to(model.device))' (image 1), pred is a tuple by default, and the true labels used for the training have a tensor format, because of this the loss cannot be calculated (image 2). To solve this I converted the tuple into a tensor which had the same size of the true labels tensor (image 1, tensor_all), and it seems that the problem is solved with that.

Image 1
![image](https://user-images.githubusercontent.com/93283899/177396890-761cbe55-b2ee-448e-a639-0707b4189a2f.png)

Image 2
![image](https://user-images.githubusercontent.com/93283899/177397354-25ff8500-0595-4800-bca7-e02ab78d89ad.png)

Now the code runs and I can see the progress of the training (image 3, below). In the image 3 is also shown some predictions (tuples) of the training. However I got a wrong prediction because all three classes (P, S and N) are getting almost the same probability, as you can see in the image 3 (below). In the image below for the first epoch I obtained the following losses:
0.673708
-0.000010
-0.000010
And if it would continue running, all the new losses would have the same value -0.000010. Do you have any idea of why I'm always getting a loss of -0.000010? In other words: Why the EQTransformer model is always predicting P arrivals (image 4)?

Image 3  
![image](https://user-images.githubusercontent.com/93283899/177397813-f8ce5347-7fbd-4b3b-b3f7-5a296bf3dbfa.png)

Image 4
![image](https://user-images.githubusercontent.com/93283899/177399246-20a81c77-6c16-4505-939d-27f7f32a4dfb.png)

**Comment by yetinam** [comment]
Hey @sfpoveda ,

EQTransformer has a slightly different output specification than PhaseNet. It outputs a tuple, consisting of a detection trace and one trace per phase. For details, check the code [here](https://github.com/seisbench/seisbench/blob/ee50da65ece48eca23a9fd35b98d200b992ce6da/seisbench/models/eqtransformer.py#L210). Crucially, it does **not** output a noise trace. Consequently, it requires binary cross-entropy loss instead of cross-entropy loss, which also explains the failure to train in your example. Have a look at our training step from the benchmark [here](https://github.com/seisbench/pick-benchmark/blob/2fae5941f632fb2146e6bea37f94eb89a2c89bc7/benchmark/models.py#L413). Please note that you'll also need to adjust the augmentations accordingly. I'd suggest to have a look at the ones used in the benchmark.
